services/workflows-service/src/business-report/business-report.controller.internal.ts

```diff
@@ -11,17 +11,157 @@ import {
   ListBusinessReportsDto,
   ListBusinessReportsSchema,
 } from '@/business-report/list-business-reports.dto';
-import { Prisma } from '@prisma/client';
+import { Business, BusinessReportStatus, Prisma } from '@prisma/client';
 import { ZodValidationPipe } from '@/common/pipes/zod.pipe';
+import axios from 'axios';
+import { env } from '@/env';
+import { CreateBusinessReportDto } from '@/business-report/dto/create-business-report.dto';
+import {
+  HookCallbackHandlerService,
+  ReportWithRiskScoreSchema,
+} from '@/workflow/hook-callback-handler.service';
+import { CustomerService } from '@/customer/customer.service';
+import { BusinessService } from '@/business/business.service';
+import { AlertService } from '@/alert/alert.service';
+import { Public } from '@/common/decorators/public.decorator';
+import { VerifyUnifiedApiSignatureDecorator } from '@/common/decorators/verify-unified-api-signature.decorator';
+import { BusinessReportHookBodyDto } from '@/business-report/dtos/business-report-hook-body.dto';
+import { BusinessReportHookSearchQueryParamsDto } from '@/business-report/dtos/business-report-hook-search-query-params.dto';
 
 @common.Controller('internal/business-reports')
 @swagger.ApiExcludeController()
 export class BusinessReportControllerInternal {
   constructor(
     protected readonly businessReportService: BusinessReportService,
     protected readonly logger: AppLoggerService,
+    protected readonly customerService: CustomerService,
+    protected readonly businessService: BusinessService,
+    protected readonly alertService: AlertService,
+    protected readonly hookCallbackService: HookCallbackHandlerService,
   ) {}
 
+  @common.Post()
+  @swagger.ApiOkResponse({ type: [String] })
+  @swagger.ApiForbiddenResponse({ type: errors.ForbiddenException })
+  async createBusinessReport(
+    @Body()
+    {
+      websiteUrl,
+      countryCode,
+      merchantName,
+      callbackUrl,
+      businessCorrelationId,
+      reportType,
+    }: CreateBusinessReportDto,
+    @CurrentProject() currentProjectId: TProjectId,
+  ) {
+    if (!merchantName && !businessCorrelationId) {
+      throw new BadRequestException('Merchant name or business id is required');
+    }
+
+    let business: Pick<Business, 'correlationId'> | undefined;
+
+    if (!businessCorrelationId && merchantName) {
+      business = await this.businessService.create({
+        data: {
+          companyName: merchantName,
+          country: countryCode,
+          website: websiteUrl,
+          projectId: currentProjectId,
+        },
+        select: {
+          correlationId: true,
+          projectId: true,
+        },
+      });
+    }
+
+    await axios.post(
```

Comment:
A suggestion, doesn't have to happen now if you think it's not needed yet - a client for the Unified-API that will abstract communications with it. Similar pattern could be used in the future, for instance a client to use an admin service or similar.

Reply:
Love the idea.